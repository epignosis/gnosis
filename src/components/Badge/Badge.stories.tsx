import React from "react";
import { Story } from "@storybook/react";
import Badge, { BadgeProps } from "./Badge";

export default {
  title: "Components/Badge",
  argTypes: {
    size: {
      control: {
        type: "select",
        options: ["md"],
      },
    },
  },
};

export const Default: Story<BadgeProps> = (args) => <Badge {...args} />;

Default.args = {
  offset: {
    right: "-8px",
    top: "0",
  },
  children: "Notifications",
  size: "md",
};
