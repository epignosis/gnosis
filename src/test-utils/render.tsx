import React, { FC, ReactElement, ComponentType } from "react";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import ThemeProvider from "../components/ThemeProvider/ThemeProvider";

const AllTheProviders: FC = ({ children }) => <ThemeProvider>{children}</ThemeProvider>;

const customRender = (ui: ReactElement, options: RenderOptions): RenderResult =>
  render(ui, { wrapper: AllTheProviders as ComponentType, ...options });

export * from "@testing-library/react";

export { customRender as render };
