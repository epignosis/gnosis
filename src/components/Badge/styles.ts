import { css, Theme, SerializedStyles } from "@emotion/react";
import { Offset, Size } from "./Badge";

const sizes = {
  md: "0.5rem",
};

export const container = (
  { badge }: Theme,
  { size, offset }: { size: Size; offset: Offset },
): SerializedStyles => {
  return css`
    display: inline-block;
    position: relative;

    &:after {
      position: absolute;
      width: ${sizes[size]};
      height: ${sizes[size]};
      content: "";
      background: ${badge.background};
      border-radius: 50%;
      top: ${offset.top};
      right: ${offset.right};
    }
  `;
};
