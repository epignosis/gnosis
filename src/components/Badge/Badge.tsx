import React, { FC } from "react";
import { SerializedStyles } from "@emotion/react";
import { container } from "./styles";

export type Size = "md";
export type Offset = { top: string; right: string };

export type BadgeProps = {
  size?: Size;
  offset?: Offset;
};

const Badge: FC<BadgeProps> = ({ size = "md", offset = { top: "0", right: "-8px" }, children }) => {
  return (
    <span css={(theme): SerializedStyles => container(theme, { size, offset })}>{children}</span>
  );
};

export default Badge;
