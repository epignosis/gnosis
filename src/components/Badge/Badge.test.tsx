import React from "react";
import { faker } from "@faker-js/faker";
import Badge from "./Badge";
import { render, screen } from "@test-utils/render";

describe("<Badge />", () => {
  it("renders correctly with badge content", () => {
    const badgeTxt = faker.lorem.word();

    render(<Badge badgeContent="5">{badgeTxt}</Badge>);
    const content = screen.getByText(badgeTxt);
    const badgeContent = screen.getByText("5");

    expect(content).toHaveTextContent(badgeTxt);
    expect(badgeContent).toHaveTextContent("5");
  });

  it("renders correctly without badge content", () => {
    const badgeTxt = faker.lorem.word();

    render(<Badge>{badgeTxt}</Badge>);
    const content = screen.getByText(badgeTxt);

    expect(content).toHaveTextContent(badgeTxt);
  });

  it("matches snapshot with badge content", () => {
    const { container } = render(
      <Badge
        id="my-id"
        className="my-class"
        size="md"
        offset={{ top: "10px", right: "10px" }}
        badgeContent="5"
      >
        Messages
      </Badge>,
    );

    expect(container).toMatchSnapshot();
  });

  it("matches snapshot without badge content", () => {
    const { container } = render(
      <Badge id="my-id" className="my-class" size="md" offset={{ top: "10px", right: "10px" }}>
        Messages
      </Badge>,
    );

    expect(container).toMatchSnapshot();
  });
});
