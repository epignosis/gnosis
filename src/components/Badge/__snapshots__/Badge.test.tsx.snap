// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> matches snapshot 1`] = `
<div>
  <span
    class="css-wwgfje-container-container"
  >
    Messages
  </span>
</div>
`;
